crates/noirc_evaluator/src/ssa_refactor/ssa_gen/context.rs

```diff
@@ -313,35 +317,50 @@ impl<'a> FunctionContext<'a> {
         // pre-loop
         let result_alloc = self.builder.insert_allocate();
         let true_value = self.builder.numeric_constant(1u128, Type::bool());
-        self.builder.insert_store(result_alloc, true_value);
+        self.builder.insert_store_with_source_location(result_alloc, true_value, location);
         let zero = self.builder.field_constant(0u128);
         self.builder.terminate_with_jmp(loop_start, vec![zero]);
 
         // loop_start
         self.builder.switch_to_block(loop_start);
         let i = self.builder.add_block_parameter(loop_start, Type::field());
         let array_length = self.builder.field_constant(array_length as u128);
-        let v0 = self.builder.insert_binary(i, BinaryOp::Lt, array_length);
+        let v0 = self.builder.insert_binary_with_source_location(
+            i,
+            BinaryOp::Lt,
+            array_length,
+            Some(location),
+        );
         self.builder.terminate_with_jmpif(v0, loop_body, loop_end);
 
         // loop body
         self.builder.switch_to_block(loop_body);
-        let v1 = self.builder.insert_array_get(lhs, i, element_type.clone());
-        let v2 = self.builder.insert_array_get(rhs, i, element_type);
-        let v3 = self.builder.insert_binary(v1, BinaryOp::Eq, v2);
-        let v4 = self.builder.insert_load(result_alloc, Type::bool());
-        let v5 = self.builder.insert_binary(v4, BinaryOp::And, v3);
-        self.builder.insert_store(result_alloc, v5);
+        let v1 = self.builder.insert_array_get_with_source_location(
+            lhs,
+            i,
+            element_type.clone(),
+            location,
+        );
+        let v2 = self.builder.insert_array_get_with_source_location(rhs, i, element_type, location);
+        let v3 =
+            self.builder.insert_binary_with_source_location(v1, BinaryOp::Eq, v2, Some(location));
+        let v4 =
+            self.builder.insert_load_with_source_location(result_alloc, Type::bool(), location);
+        let v5 =
+            self.builder.insert_binary_with_source_location(v4, BinaryOp::And, v3, Some(location));
+        self.builder.insert_store_with_source_location(result_alloc, v5, location);
         let one = self.builder.field_constant(1u128);
-        let v6 = self.builder.insert_binary(i, BinaryOp::Add, one);
+        let v6 =
+            self.builder.insert_binary_with_source_location(i, BinaryOp::Add, one, Some(location));
```

Comment:
Can we change this to `self.builder.set_location(location).insert_binary(i, BinaryOp::Add, one)`?

Reply:
This is not clear to me; 
if we duplicate the builder, it will write the instruction on its own function, and not on the one from self.builder.
if we do not duplicate the builder, we would need to set the location every time it might have been overwritten, which seems to be error prone.

Reply:
Finally I do not duplicate and we update the location when we have one